tests/rptest/tests/topic_creation_test.py

```diff
@@ -362,6 +364,94 @@ def test_min_rf_log(self):
+        super(CreateTopicsResponseTest,
+              self).__init__(test_context=test_context)
+        self.kcl_client = RawKCL(self.redpanda)
+        self.admin = Admin(self.redpanda)
+
+    # we don't really care about the name aside from its not being random
+    # so just construct it from the partition count and replication factor
+    def create_topics(self, p_cnt, r_fac, n=1):
+        topics = []
+        for i in range(0, n):
+            topics.append({
+                'name': f"foo-{p_cnt}-{r_fac}-{i}",
+                'partition_count': p_cnt,
+                'replication_factor': r_fac
+            })
+
+        return self.kcl_client.create_topics(6, topics=topics)
+
+    def get_np(self, tp):
+        return tp['NumPartitions']
+
+    def get_rf(self, tp):
+        return tp['ReplicationFactor']
+
+    def get_ec(self, tp):
+        return tp['ErrorCode']
+
+    def get_configs(self, tp):
+        return tp['Configs']
+
+    def get_config_by_name(self, tp, name):
+        cfgs = self.get_configs(tp)
+        return next((cfg for cfg in cfgs if cfg['Name'] == name), None)
+
+    @cluster(num_nodes=3)
+    @matrix(
+        partition_count=[3, -1],
+        replication_factor=[3, -1],
+    )
+    def test_create_topic_responses(self, partition_count, replication_factor):
+        """
+        Validates that create_topic responses are populated with real values when
+        default placeholders are supplied in the request
+        """
+
+        cfg = self.admin.get_cluster_config()
+        expected_np = partition_count if partition_count > 0 else cfg[
+            'default_topic_partitions']
+        expected_rf = replication_factor if replication_factor > 0 else cfg[
+            'default_topic_replications']
+
+        topics = self.create_topics(partition_count, replication_factor, 3)
+        for topic in topics:
+            assert self.get_np(topic) == expected_np
+            assert self.get_rf(topic) == expected_rf
+            assert self.get_ec(topic) == self.SUCCESS_EC
+
+        topics = self.create_topics(partition_count, replication_factor, 3)
+        for topic in topics:
+            assert self.get_np(
+                topic) == expected_np, f"Got {self.get_np(topic)}"
+            assert self.get_rf(topic) == expected_rf
+            assert self.get_ec(topic) == self.TOPIC_EXISTS_EC
+
+    @cluster(num_nodes=3)
+    def test_create_topic_response_configs(self):
+        """
+        Validates that configs returned in create_topics responses are
+          a. qualified with an appropriate "source"
+          b. serialized correctly
+        """
+
+        topics = self.create_topics(1, 1)
+        for topic in topics:
+            cleanup_policy = self.get_config_by_name(topic, 'cleanup.policy')
+            assert cleanup_policy is not None
+            assert cleanup_policy['Value'] == self.DEFAULT_CLEANUP_POLICY
+            assert cleanup_policy['Source'] == self.DEFAULT_CONFIG_SOURCE
```

Comment:
nit: Add a string message for these asserts (and the ones above) so when they fail we have a nice message, e.g.

`assert cleanup_policy['Value'] == self.DEFAULT_CLEANUP_POLICY, f'{cleanup_policy["Value"]} != {self.DEFAULT_CLEANUP_POLICY}`

Reply:
ya no problem

Reply:
done